src/CoreWCF.Primitives/src/CoreWCF/Security/EncryptedKey.cs

```diff
@@ -0,0 +1,537 @@
+            _wsuId = reader.GetAttribute(XD.XmlEncryptionDictionary.Id, XD.UtilityDictionary.Namespace) ?? SecurityUniqueId.Create().Value;
+            _mimeType = reader.GetAttribute(MimeTypeAttribute, null);
+            _type = reader.GetAttribute(TypeAttribute, null);
+            ReadAdditionalAttributes(reader);
+            reader.Read();
+
+            if (reader.IsStartElement(EncryptionMethodElement.ElementName, NamespaceUri))
+            {
+                _encryptionMethod.ReadFrom(reader);
+            }
+
+            if (_tokenSerializer.CanReadKeyIdentifier(reader))
+            {
+                XmlElement xml = null;
+                XmlDictionaryReader localReader;
+
+                if (this.ShouldReadXmlReferenceKeyInfoClause)
+                {
+                    // We create the dom only when needed to not affect perf.
+                    XmlDocument doc = new XmlDocument();
+                    xml = (doc.ReadNode(reader) as XmlElement);
+                    localReader = XmlDictionaryReader.CreateDictionaryReader(new XmlNodeReader(xml));
+                }
+                else
+                {
+                    localReader = reader;
+                }
+
+                try
+                {
+                    this.KeyIdentifier = _tokenSerializer.ReadKeyIdentifier(localReader);
+                }
+                catch (Exception e)
+                {
+                    // In case when the issued token ( custom token) is used as an initiator token; we will fail 
+                    // to read the keyIdentifierClause using the plugged in default serializer. So We need to try to read it as an XmlReferencekeyIdentifierClause 
+                    // if it is the client side.
+
+                    if (Fx.IsFatal(e) || !this.ShouldReadXmlReferenceKeyInfoClause)
+                    {
+                        throw;
+                    }
+
+                    _keyIdentifier = ReadGenericXmlSecurityKeyIdentifier(XmlDictionaryReader.CreateDictionaryReader(new XmlNodeReader(xml)), e);
+                }
+            }
+
+            reader.ReadStartElement(CipherDataElementName, EncryptedType.NamespaceUri);
+            reader.ReadStartElement(CipherValueElementName, EncryptedType.NamespaceUri);
+            if (maxBufferSize == 0)
+                ReadCipherData(reader);
+            else
+                ReadCipherData(reader, maxBufferSize);
+            reader.ReadEndElement(); // CipherValue
+            reader.ReadEndElement(); // CipherData
+
+            ReadAdditionalElements(reader);
+            reader.ReadEndElement(); // OpeningElementName
+            this.State = EncryptionState.Read;
+        }
+
+        private SecurityKeyIdentifier ReadGenericXmlSecurityKeyIdentifier(XmlDictionaryReader localReader, Exception previousException)
+        {
+            throw new NotImplementedException();
```

Comment:
It looks like you copied the one from the wcf repo but from the dotnet-svcutil-lib project. That is a significantly gutted implementation and isn't used when WCF is running. Can you re-port this file from .NET Framework as it's going to be significantly broken otherwise. Do you know if you copied any other files from that folder?

Reply:
updated above method.. I am using https://github.com/microsoft/referencesource to move code across..